ami/utils/tests.py

```
@@ -32,3 +35,33 @@ def test_extract_timestamps(self):
+        result = extract_error_message_from_response(mock_response)
+        self.assertIn("error: Invalid input", result)
+
+        # Test fallback to text when JSON fails
+        mock_response.json.side_effect = ValueError("No JSON")
+        mock_response.text = "Service unavailable"
+        result = extract_error_message_from_response(mock_response)
+        self.assertIn("Response text: Service unavailable", result)
+
+        # Test fallback to raw bytes when text access fails
+        mock_response.json.side_effect = ValueError("404 Not Found: Could not fetch image")
+        mock_response.text = property(lambda self: (_ for _ in ()).throw(Exception("text error")))
```

Ruff (0.14.2)

Unused lambda argument: `self`

(ARG005)